### ISSUE TITLE
Rule Request: setter functions variable name to value

### ISSUE THREAD

**Issue by michaeljota** [eslint/eslint#9719]
**Please describe what the rule should do:**
Enforce the variable of the set `functions` to `value`.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const person = {
  set name(name) {
    this._name = name;
  }
}

const obj = {
  set shape(shape) {
    this._shape = shape;
  }
}
```

***Should pass with:***
```js
const person = {
  set name(value) {
    this._name = value;
  }
}

const obj = {
  set shape(value) {
    this._shape = value;
  }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
I think this would be important to have the same name in all setter functions. I look up for this, but found nothing. `value` is suggested as it is wide used as the setter variable name, but it is actually not enforced, and could be named anything else.

**Comment by not-an-aardvark** [comment]
Could you clarify why you want the argument of all setter functions to be called `value`? I'm not sure I understand the purpose of doing that; it seems like some setter functions could benefit from more specific names.

**Comment by michaeljota** [comment]
The name `value` is a common name in setters in other languages, and its using is extended in JS as well. So, using a rule like this would make that name standard inside a project.

**Comment by not-an-aardvark** [comment]
Thanks for clarifying. Personally, I don't think this rule would be a good fit for ESLint core, since I think it's a fairly uncommon preference. However, you could enforce this on your own project using the [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax) rule:

```json
{
  "rules": {
    "no-restricted-syntax": [
      "error",
      {
        "selector": ":matches(MethodDefinition, Property)[kind='set'][value.params.0.name!='value']",
        "message": "Setter parameters should be called 'value'."
      }
    ]
  }
}
```

**Comment by michaeljota** [comment]
Thanks! That looks superb!

**Comment by not-an-aardvark** [comment]
Closing because this can be addressed by `no-restricted-syntax`.
